### DISCUSSION TITLE
Template Inheritance

### DISCUSSION THREAD

**Discussion by vanadium23** [SilentVoid13/Templater#1005] [Ideas & Feature Requests]
After year of using templater, I've found out that 25% of my tempates are basics templates with a little addition by domain. First, I was using conditional to do so, than move to `include`. But it's hard to maintain across 15 templates to have consistent templates. May be we can add feature `extends` like in jinja2?

**Comment by welpdx** [comment]
can you explain exactly what feature `extends` is? I dont use `jinja2`. Thanks

**Reply by vanadium23** [reply]
No worries, you can read it in official documentation:
https://jinja.palletsprojects.com/en/2.11.x/templates/#child-template

Or as a simple example, I want to make a base.md template like:
```markdown
---
date: <% tp.date.now %>
<% tp.block.fronmatter %>
<% tp.block.end %>
---

# <% tp.file.title %>
```

And in other templates choose base.md as tempate
```markdown
<% tp.extends([[base]]) %>

<% tp.block.frontmatter %>
tags: [project]
<% tp.block.end %>
```

**Reply by welpdx** [reply]
I found a [yt video](https://www.youtube.com/watch?v=VS-1vD81Pdc) that summarizes the documentation. Pretty neat idea. I feel like the existing code for changing cursor location in templater can be used for this.

**Reply by vanadium23** [reply]
Can you provide an example?

**Reply by welpdx** [reply]
I'm not a programmer, but as I can understand it, templater has this function called [tp.file.cursor_append(content: string)](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilecursor_appendcontent-string), which basically replaces the <%* tp.file.cursor_append(string) %> with string when a template is created. 

The same kind of idea can be used for your `extend` idea, for a possible future PR. 

See the code here: https://github.com/SilentVoid13/Templater/blob/487805b5ad1fd7fbc145040ed82b4c41fc2c48e2/src/core/functions/internal_functions/file/InternalModuleFile.ts#L110

**Comment by sohanglal** [comment]
> After year of using templater, I've found out that 25% of my tempates are basics templates with a little addition by domain. First, I was using conditional to do so, than move to `include`. But it's hard to maintain across 15 templates to have consistent templates. May be we can add feature `extends` like in jinja2?

`tp.file.include()`
